docs/guides/advanced/cli.md

```diff
@@ -4,7 +4,230 @@ title: Using the CLI
+
+To pass array parameters to `coda execute`, use a comma separated string. For example, [1,2,3] should be passed with this format:
+
+```sh
+coda execute src/pack.ts GetAverage "1,2,3"
+```
+
+
+### Running Syncs
+
+The above example shows how to execute a regular Pack formula. Executing a sync is almost identical:
+
+```sh
+coda execute path/to/pack.ts <sync name> [params..]
+```
+
+So for example, if you had a sync called Items, that took a start date as a parameter, you would execute this as:
+
+```sh
+coda execute path/to/pack.ts Items "2020-12-15"
+```
+
+This will execute your sync formula repeatedly until there are no more results, and print the output array of all result objects to the terminal. See the [Sync tables guide][sync_tables] for more information about how and why sync formulas are invoked repeatedly for paginated results.
+
+To run a sync for a dynamic sync table, use the `--dynamicUrl` parameter to specify which URL to sync from.
+
+```sh
+coda execute path/to/pack.ts Items --dynamicUrl=https://example.com/api/table
+```
+
+
+### Fetching {: #fetch}
+
+By default, `coda execute` will use a mock fetcher for any http requests that your formulas make. If you wish to actually make http requests, use the `--fetch` flag, for example:
+
+```sh
+coda execute --fetch src/pack.ts GetPrice "widgets"
+```
+
+Your http requests will commonly require authentication in order to succeed, which the coda execute utility supports. See the [Authentication section](#authentication) for more information on how to set this up.
+
+
+## Authentication {: #authentication}
+
+The SDK will help you set up authentication in your development environment so that you can execute Pack formulas with authentication applied to them. This allows you to run your code end-to-end including making fetcher requests to external APIs.
+
+The `coda auth` utility is used to set up authentication for a Pack. Run `coda auth --help` at any time for a refresher on how to use the utility. Mostly, it’s as simple as running
+
+```sh
+coda auth path/to/pack.ts
+```
+
+The utility will inspect your Pack definition to see what kind of authentication you have defined, and then it will prompt you to provide in the console the necessary token(s) or other parameters required by your authorization type. If you are using `OAuth2`, after you provide the necessary configuration info, it will launch an OAuth flow in your browser. The resulting credentials you provide will be stored in a file `.coda-credentials.json` in the same directory as your Pack definition.
+
+The credentials will be automatically applied to your fetch requests when you execute a Pack from the CLI or a test. For more information, see the sections on [Using the --fetch option](#fetch) and [Integration tests][integration].
```

Comment:
Similarly, can probably simplify and remove the reference to --fetch now.

Reply:
Done.